packages/matter.js/src/protocol/interaction/InteractionClient.ts

```diff
@@ -360,34 +415,101 @@ export class InteractionClient {
         return this.withMessenger<void>(async messenger => {
-            logger.debug(`Sending subscribe request: ${attributeRequests.map(path => resolveAttributeName(path)).join(", ")}`);
+            logger.debug(`Sending subscribe request: attributes: ${attributeRequests.map(path => resolveAttributeName(path)).join(", ")} and events: ${eventRequests.map(path => resolveEventName(path)).join(", ")}`);
             const { report, subscribeResponse: { subscriptionId } } = await messenger.sendSubscribeRequest({
                 attributeRequests,
-                keepSubscriptions: true,
+                eventRequests,
+                keepSubscriptions,
                 minIntervalFloorSeconds,
                 maxIntervalCeilingSeconds,
-                isFabricFiltered: true,
-            }); // TODO: also initialize all values
+                isFabricFiltered,
+            });
 
             const subscriptionListener = (dataReport: DataReport) => {
-                if (!Array.isArray(dataReport.attributeReports) || !dataReport.attributeReports.length) {
+                if (
+                    (!Array.isArray(dataReport.attributeReports) || !dataReport.attributeReports.length) &&
+                    (!Array.isArray(dataReport.eventReports) || !dataReport.eventReports.length)
+                ) {
                     logger.debug('Subscription result empty');
                     return;
                 }
-                const values = normalizeAndDecodeReadAttributeReport(dataReport.attributeReports);
-
-                values.forEach(data => {
-                    const { path: { endpointId, clusterId, attributeId }, value, version } = data;
-                    if (value === undefined) throw new MatterFlowError('Subscription result reporting undefined value not specified');
-                    this.subscribedLocalValues.set(attributePathToId({ endpointId, clusterId, attributeId }), { value, version });
-                    listener?.(data);
-                });
+                const { attributeReports, eventReports } = dataReport;
+
+                if (attributeReports !== undefined) {
+                    const attributeValues = normalizeAndDecodeReadAttributeReport(attributeReports);
+
+                    attributeValues.forEach(data => {
+                        const { path: { endpointId, clusterId, attributeId }, value, version } = data;
+                        if (value === undefined) throw new MatterFlowError('Subscription result reporting undefined value not specified');
```

Comment:
This error is a little confusing, is it "Attribute subscription event with no payload"?

Reply:
rephrased